p_soc_auto_base/utils.py

```diff
@@ -37,6 +39,48 @@
 LOGGER = logging.getLogger('django')
 
 
+def get_absolute_admin_change_url(
+        admin_view, obj_pk, obj_anchor_name=None, root_url=None):
+    """
+    build and return an absolute `URL` for a `Django admin change` page
+
+    See `Reversing admin URLs
+    <https://docs.djangoproject.com/en/2.2/ref/contrib/admin/#reversing-admin-urls>`__.
+
+    :arg str admin_view: the name of the admin view
+    :arg int obj_pk: the primary key of the object in the view
+    :arg str obj_anchor_name: the string that should be placed inside
+        an <a>$obj_anchor_name</a> `HTML` tag; default is `None` and
+        in that case the function assumes that the HTML tag will be
+        <a></a>
+
+    :arg str root_url: an `URL` fragment to be placed before the
+        `Django admin` object path
+
+        It is supposed to look something like
+        '<a href="http://server:port/'. If this argument is set to `None`,
+        it will be populated from specific `Django` settings and the
+        value returned by :meth:`socket.getfqdn`
+
+    """
+    if root_url is None:
+        root_url = (
+            f'<a href="{settings.SERVER_PROTO}://{socket.getfqdn()}:'
+            f'{settings.SERVER_PORT}/')
+
+    if obj_anchor_name is None:
+        tail_url = '"></a>'
+    else:
+        tail_url = f'">{obj_anchor_name}</a>'
+
+    try:
+        admin_path = reverse(admin_view, args=(obj_pk,))
+    except NoReverseMatch as err:
+        return f'cannot resolve URL path for view {admin_view}. Error: {err}'
```

Comment:
It seems odd that this method, which a user expects to return a string that is a url, would return a string which is an error message when it fails. How is the output being used, will it simply be displayed? I worry the user will try to go to the return value and get some weird error about not being able to resolve the address instead of the intended message.
Is it possible to simply log this message and return `''` or `None` instead?

Reply:
the return is used in an email message and we want the reader to do something about an Orion node not being resolved for some reason or other